ibm/service/power/resource_ibm_pi_volume_bulk.go

```diff
@@ -0,0 +1,527 @@
+							Computed:    true,
+							Description: "Indicates master volume name",
+							Type:        schema.TypeString,
+						},
+						Attr_MirroringState: {
+							Computed:    true,
+							Description: "Mirroring state for replication enabled volume",
+							Type:        schema.TypeString,
+						},
+						Attr_PrimaryRole: {
+							Computed:    true,
+							Description: "Indicates whether 'master'/'auxiliary' volume is playing the primary role.",
+							Type:        schema.TypeString,
+						},
+						Attr_ReplicationStatus: {
+							Computed:    true,
+							Description: "The replication status of the volume.",
+							Type:        schema.TypeString,
+						},
+						Attr_ReplicationSites: {
+							Computed:    true,
+							Description: "List of replication sites for volume replication.",
+							Elem:        &schema.Schema{Type: schema.TypeString},
+							Type:        schema.TypeList,
+						},
+						Attr_ReplicationType: {
+							Computed:    true,
+							Description: "The replication type of the volume 'metro' or 'global'.",
+							Type:        schema.TypeString,
+						},
+						Attr_VolumeID: {
+							Computed:    true,
+							Description: "The unique identifier of the volume.",
+							Type:        schema.TypeString,
+						},
+						Attr_VolumeStatus: {
+							Computed:    true,
+							Description: "The status of the volume.",
+							Type:        schema.TypeString,
+						},
+						Attr_WWN: {
+							Computed:    true,
+							Description: "The world wide name of the volume.",
+							Type:        schema.TypeString,
+						},
+					},
+				},
+				Type: schema.TypeList,
+			},
+		},
+	}
+}
+
+func ResourceIBMPIVolumeBulkValidator() *validate.ResourceValidator {
+	validateSchema := make([]validate.ValidateSchema, 0)
+	validateSchema = append(validateSchema,
+		validate.ValidateSchema{
+			Identifier:                 "pi_affinity",
+			ValidateFunctionIdentifier: validate.ValidateAllowedStringValue,
+			Type:                       validate.TypeString,
+			Required:                   true,
+			AllowedValues:              "affinity, anti-affinity"})
+	IBMPIVolumeBulkResourceValidator := validate.ResourceValidator{
+		ResourceName: "ibm_pi_volume",
+		Schema:       validateSchema}
+	return &IBMPIVolumeBulkResourceValidator
+}
+
+func resourceIBMPIVolumeBulkCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	name := d.Get(Arg_VolumeName).(string)
+	size := int64(d.Get(Arg_VolumeSize).(float64))
+	var shared bool
+	if v, ok := d.GetOk(Arg_VolumeShareable); ok {
+		shared = v.(bool)
+	}
+	cloudInstanceID := d.Get(Arg_CloudInstanceID).(string)
+	body := &models.MultiVolumesCreate{
+		Name:      &name,
+		Shareable: &shared,
+		Size:      &size,
+	}
+	if v, ok := d.GetOk(Arg_Count); ok {
+		count := v.(int)
+		body.Count = int64(count)
+	} else {
+		body.Count = 1
+	}
```

Comment:
You could just make the default value 1 in the schema instead of having the GetOk and else block

Reply:
updated default